resources/js/components/FormField.vue

```diff
@@ -38,9 +38,20 @@
 			registerDependencyWatchers(root) {
 				root.$children.forEach(component => {
 					if (this.componentIsDependency(component)) {
-
-						component.$watch('value', (value) => {
-							this.dependencyValues[component.field.attribute] = value;
+						let attrname = this.field.watchable.name,
+							attrpath = this.field.watchable.path;
+
+						component.$watch(attrname, (value) => {
+							let result = null
+							if (attrpath) {
```

Comment:
divide responsibility of this `if` statement to a function always called independently of an `if`, e.g

```javascript
component.$watch(attrname, (value) => {
    // findComponentResult (or whatever it may be called) is now responsible
    let result = this.findComponentValue(value, attrpath);

    this.dependencyValues[component.field.attribute] = result;
    this.updateDependencyStatus();
}, {immediate: true})
```

```javascript
this.findComponentValue(value, attrpath) {
    if(attrpath)
        return (value && typeof value === 'object'
            && value.hasOwnProperty(attrpath))
            ? value[attrpath]
            : null
    }
    return value;
});
```